core/src/main/java/org/eclipse/hono/adapter/AdapterConfig.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2016 Red Hat
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Red Hat - initial creation
+ */
+
+package org.eclipse.hono.adapter;
+
+import io.vertx.core.Vertx;
+import org.eclipse.hono.config.HonoClientConfigProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Minimum configuration for protocol adapters
+ */
+@Configuration
+public abstract class AdapterConfig {
```

Comment:
I started the MQTT adapter adapting the same REST code so I used the factory for returning the verticle. I guess that it's not necessary. The verticle instance, specific for each adapter, could be something not injectable but just a well defined field in the adapter application. As you see the Config class in our REST and MQTT adapter just have the factory that sounds not mandatory to me. Our is just a possible implementation using a factory. What do you think ?